src/window/system_clock.rs

```diff
@@ -18,8 +18,15 @@ use super::{Clock, ClockConfig};
 ///   Config object. Pass this as the `clock_config` parameter to
 ///   your windowing operator.
 #[pyclass(module="bytewax.window", extends=ClockConfig)]
+#[derive(Clone, Copy)]
```

Comment:
Why are these suddenly needed on our config types?

Reply:
So, this is due to the new `ClockBuilder` trait, which consumes `self`.

Previously in the `build_clock_builder` function we were manually cloning all the fields in the config, and passing them to the builder. With the new trait we clone the whole config instead (required by the `FromPyObject` trait, used in `extract`). So instead of implicitly requiring all the fields in the config to be `Clone`, we require the config itself to be cloneable.
Since we end up cloning all the fields in the end, I thought having this requirement here wouldn't do much of a difference.

This change is not needed for this PR, I can easily revert it, remove the `ClockBuilder` trait, and implement the `builder` function on the struct like it was previously. It's a refactoring I did along the way to make the pattern we use more explicit, and I feel it makes the code more readable, but it works the other way too.